index.js

```diff
@@ -6,7 +6,11 @@ module.exports = function (CustomEmitter) {
   var emit = Emitter.prototype.emit
 
   function onevent (packet) {
-    var args = packet.data || []
+    var args = [];
+
+    if (packet.data && packet.data.constructor === Array) {
```

Comment:
For those `packet.data` which is not an Array, what are them? Any reason that we should ignore them?